deployment/action.yml

```diff
@@ -0,0 +1,94 @@
+name: "Setops deployment"
+description: "Creates and activates releases for multiple apps within a SetOps stage, performs a predeploy command, waits for the releases to become healthy"
+
+inputs:
+  setops-organization:
+    description: The SetOps organization
+    required: false
+  setops-project:
+    description: The SetOps project name
+    required: true
+  setops-username:
+    description: The SetOps username, usually obtained via secrets.SETOPS_USER
+    required: true
+  setops-password:
+    description: The SetOps password, usually obtained via secrets.SETOPS_PASSWORD
+    required: true
+  image-digest:
+    description: The image digest that was returned when pushing the image to the SetOps registry
+    required: true
+  stage:
+    description: The SetOps stage for the deployment, e.g. production
+    required: true
+  apps:
+    description: List of apps in JSON format, e.g. '["web", "clock", "worker"]'
+    required: true
+  number-of-retries-to-wait-for-successful-deployment:
+    description: Max. number of times to wait for a succesful deployment (we sleep for 5 seconds between tries, so 12 equals 1 minute)
+    required: false
+    default: "24"
+  predeploy-command:
+    description: A predeploy command to be run before activating the releases, e.g. for migrating the database
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Setops"
+      uses: setopsco/setup-setops@v3
+      with:
+        setops_organization: ${{ inputs.setops-organization }}
+        setops_username: ${{ inputs.setops-username }}
+        setops_password: ${{ inputs.setops-password }}
+    - name: "Setops deployment"
+      run: |
+        shopt -s expand_aliases
+        alias sos='setops -p ${{ inputs.setops-project }} -s ${{ inputs.stage }}'
+
+        declare -a apps
+        for app in $(echo '${{ inputs.apps }}' | jq -r '.[]'); do
+          apps+=($app)
+        done
+
+        sos changeset:discard || true
+
+        declare -A apps_with_release_id
+
+        for app in ${apps[@]}; do
+          apps_with_release_id[$app]=$(sos --app $app release:create ${{ inputs.image-digest }} | grep -o 'ReleaseID.*' | grep -o '[0-9].*')
+        done
+
+        echo "Changeset for creating releases"
+        sos changeset:commit
+
+        predeploy_command="${{ inputs.predeploy-command }}"
```

Comment:
Could this include double quotes? If so, should they be escaped via Github Action syntax?